tool/release/createnotes/Note.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.vaticle.dependencies.tool.release.createnotes
+
+import com.eclipsesource.json.Json
+import com.eclipsesource.json.JsonArray
+import com.eclipsesource.json.JsonObject
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelBug
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelFeature
+import com.vaticle.dependencies.tool.release.createnotes.Constant.github
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelPrefix
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelRefactor
+
+class Note {
+    enum class Type { FEATURE, BUG, REFACTOR, OTHER }
+
+    private val title: String
+    private val goal: String?
+    private val type: Type
+
+    private constructor(title: String, goal: String?, type: Type) {
+        this.title = title
+        this.goal = goal
+        this.type = type
+    }
+
+    fun toMarkdown():  String {
+        return "- **$title**\n  ${goal ?: ""}"
+    }
+
+    companion object {
+        fun fromGithubPR(pr: JsonObject): Note {
+            return Note(
+                title = pr.get("title").asString(),
+                goal = getPRGoal(pr.get("body").asString()),
+                type = getPRType(pr.get("labels").asArray())
+            )
+        }
+
+        private fun getPRType(labels: JsonArray): Type {
+            val types = labels
+                .map { e -> e.asObject().get("name").asString() }
+                .filter { e -> e.startsWith(labelPrefix) }
+            val type =
```

Comment:
It is possible for a PR to have multiple type labels (eg., feature & refactor), but the note should belong to one type.